CHANGELOG.md

```diff
@@ -1,5 +1,20 @@
 :warning: **Important**: If you are upgrading to version 2.3.0 or later and have firewall rules or network configuration that blocks any unknown traffic by default, you need to update your configuration to allow connections to the new DNS names and IP addresses. Please refer to this [changelog](#230-january-23-2023) for more details.
 
+2.8.0 (In Progress)
+===================
+
+Changes
+-------
+
+- Enhanced error reporting from backend services. Signaling errors emitted by `Device` and `Call` objects should now be more descriptive. If your application logic relied on parsing generic error code `31005`, please consider reading through our API documentation and adjusting your application logic to anticipate a wider and more descriptive range of error codes.
+  - Some new error codes included within this release are:
+    - **ClientErrors**
+      - NotFound: `31404`
+      - TemporarilyUnavailable: `31480`
+      - BusyHere: `31486`
+    - **SIPServerErrors**
+      - Decline: `31603`
+
```

Comment:
- Added new errors emitted by `Device` and `Call` objects. Previously, these errors are all part of a generic error code `31005`. With this new release, the following errors now have their own error codes. Please see this [page](https://www.twilio.com/docs/api/errors) for more details about each error.
  - **ClientErrors**
    - NotFound: `31404`
    - TemporarilyUnavailable: `31480`
    - BusyHere: `31486`
  - **SIPServerErrors**
    - Decline: `31603`

_IMPORTANT: If your application relies on listening to the generic error code 31005 when any of the errors above happen, you need to update your error listeners to also listen for the new error codes._